src/main/java/ivanov/springbootintro/dto/BookDto.java

```diff
@@ -1,14 +1,22 @@
 package ivanov.springbootintro.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookDto {
+    @NotNull
     private Long id;
+    @NotNull

```

Comment:
For String is better to use @NotBlank or @NotEmpty
https://www.baeldung.com/java-bean-validation-not-null-empty-blank